src/Transport/Topology/Topologies/ForwardingTopologySectionManager.cs

```diff
@@ -89,30 +98,33 @@ public TopologySectionInternal DetermineResourcesToCreate(QueueBindings queueBin
             };
         }
 
-        public TopologySectionInternal DeterminePublishDestination(Type eventType)
+        public async Task<TopologySectionInternal> DeterminePublishDestination(Type eventType)
         {
-            return publishDestinations.GetOrAdd(eventType, t =>
+            TopologySectionInternal destination;
```

Comment:
@danielmarbach I couldn't use `ConcurrentDictionary.GetOrAdd`. Can you poke any holes in this? Thanks.